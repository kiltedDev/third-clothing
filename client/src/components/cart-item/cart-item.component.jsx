import React from 'react';

import {
  CartItemContainer,
  CartItemImage,
  CartItemDetails,
  CartItemName,
  CartItemPrice
} from './cart-item.styles';

export const CartItem = ({ item: { imageUrl, price, name, quantity } }) => {
  return (
    <CartItemContainer className='cart-item'>
      <CartItemImage src={ imageUrl } alt={ name } />
      <CartItemDetails className='item-details' >
        <CartItemName className='name' children={ name } />
        <CartItemPrice className='price' children={ `${quantity} x $${price}` } />
    </CartItemDetails>
  </CartItemContainer>
  )
}

export default React.memo( CartItem );
