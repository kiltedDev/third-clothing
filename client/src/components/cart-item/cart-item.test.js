import { shallow } from 'enzyme';
import React from 'react';
import { CartItem } from './cart-item.component';

describe('CartItem', () => {
  let wrapper;

  beforeAll(() => {
    const mockProps = {
      item: {
        imageUrl: 'www.images.com/cool-image.png',
        quantity: 3,
        price: 25,
        name: 'Really Cool Coat'
      }
    }

    wrapper = shallow(<CartItem { ...mockProps } /> );
  })

  it('renders a cart item',() => {
    expect( wrapper.debug() ).toMatchSnapshot();
  })

  it('renders the correct name for the item',() => {
    const itemName = wrapper.find('CartItemName').text()
    expect( itemName ).toEqual('Really Cool Coat');
  })

  it('renders the correct price for the item',() => {
    const itemPrice = wrapper.find('CartItemPrice').text()
    expect(itemPrice).toEqual('3 x $25');
  })
})
