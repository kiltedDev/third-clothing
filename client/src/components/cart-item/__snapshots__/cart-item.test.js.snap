// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders a cart item 1`] = `
"<CartItemContainer className=\\"cart-item\\">
  <styled.img src=\\"www.images.com/cool-image.png\\" alt=\\"Really Cool Coat\\" />
  <styled.div className=\\"item-details\\">
    <CartItemName className=\\"name\\">
      Really Cool Coat
    </CartItemName>
    <CartItemPrice className=\\"price\\">
      3 x $25
    </CartItemPrice>
  </styled.div>
</CartItemContainer>"
`;
